src/components/CreditCardForm.js

```diff
@@ -0,0 +1,141 @@
+import React, { useState } from "react";
+
+function CreditCardForm(props) {
+  const userDefault = {
+    firstName: {
+      value: "",
+    },
+    surname: {
+      value: "",
+    },
+    cardNumber: {
+      value: "",
+      error: "",
+    },
+    expirationDate: {
+      value: "",
+    },
+  };
+
+  const [user, setUser] = useState(userDefault);
+  const [creditCardType, setCreditCardType] = useState("");
+
+  function checkCardNumber(value) {
+    if (value.length >= 13 && value.length <= 16) {
+      let newNum = value.padStart(16, "0");
+      let sum = 0;
+      let newSum = 0;
+      for (let i = 0; i < newNum.length; i++) {
+        if (i === 0) {
+          sum = (parseInt(newNum[i]) * 2).toString();
+        } else if (i % 2 === 0) {
+          sum = sum + (parseInt(newNum[i]) * 2).toString();
+        } else {
+          sum = sum + (parseInt(newNum[i]) * 1).toString();
+        }
+      }
+      for (let j = 0; j < sum.length; j++) {
+        newSum = newSum + parseInt(sum[j]);
+      }
+      let newSumStr = newSum.toString();
+      if (newSumStr.charAt(newSumStr.length - 1) !== "0") {
+        throw new Error("Wrong card number!");
+      } else {
+        console.log("Correct number!");
+      }
+    }
+  }
```

Comment:
Raczej wrzuciłbym tą funkcję do osobnego pliku np. `creditcard.helper.js` (bo jest spora i skomplikowana) - jeśli ktoś będzie chciał się z nią zapoznać to zajrzy do pliku.